File: cpp/linked_list.cpp
#include <iostream>

using namespace std;

class Node{
public:
    int data;
    Node* next;
};

void print_list(Node* n){
    while(n!=nullptr){
        cout<<n->data<<endl;
        n = n->next;
    }
}

int main(){
    Node* head = nullptr;
    Node* second = nullptr;
    Node* third = nullptr;

    head = new Node();
    second = new Node();
    third = new Node();

    head->data = 1;
    head->next = second;
    second->data = 2;
    second->next = third;
    third->data = 3;
    third->next = nullptr;

    print_list(head);

    return 0;
}

File: linked_list.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LinkedList:

    def __init__(self):
        self.head = None

    def print_list(self):
        current = self.head
        while(current):
            print(current.data, end=' ')
            current = current.next

if __name__=='__main__':

    linked_list = LinkedList()

    linked_list.head = Node(1)
    second = Node(2)
    third = Node(3)
    
    second.next = third
    linked_list.head.next = second
    #next of third is already None

    linked_list.print_list()

